build_tools/scripts/update_submodules.sh

```diff
@@ -37,8 +37,29 @@ if [ -n "$git_status" ]; then
   exit 1
 fi
 
-echo "Pulling latest from remote"
-git pull
+master_url="$(git remote get-url master)"
+upstream_url="https://github.com/google/iree.git"
+if [ ${master_url} == ${upstream_url} ]
+then
+  echo "Pulling latest from remote"
+  git pull
+else
+  for remote in `git remote -v | awk -F\  '{print $1}' | sort | uniq`
+  do
+    remote_url="$(git remote get-url ${remote})"
+    if [ ${remote_url} == ${upstream_url} ]
+    then
+      upstream=${remote}
+      break
+    fi
+  done
+
+  echo "Fetching" ${upstream}
+  git fetch ${upstream}
+
+  echo "Merging fork master with " ${upstream}
+  git merge ${upstream}/master
+fi
```

Comment:
I think this whole thing can be replaced with

`git pull https://github.com/google/iree.git`

Reply:
I'll try that if that works. But I using this : https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork

Reply:
Yeah I think fetching separately is a good idea usually, but having to do the manual search through for the remote name that matches the URL is painful. The nice thing about pull here is you can do it based on a URL. That tutorial also recommends merging. I think the GitHub tutorials in general are way too fond of merging and IMO merge commits kind of suck.